repair/repair.cc

```diff
@@ -2143,28 +2144,27 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
         co_await tmap.for_each_tablet([&] (locator::tablet_id id, const locator::tablet_info& info) -> future<> {
             auto range = tmap.get_token_range(id);
             auto& replicas = info.replicas;
+
+            if (primary_replica_only) {
+                auto primary_replica = select_primary_ranges_within_dc ?
+                        tmap.get_primary_replica_within_dc(id, erm->get_topology(), mydc) :
+                        tmap.get_primary_replica(id);
+                if (primary_replica.host == myhostid) {
+                    metas.push_back(repair_tablet_meta{id, range, myhostid, primary_replica.shard, replicas});
+                }
+                return make_ready_future<>();
+            }
+
```

Comment:
We could share the meta.push_back with the non primary_replica_only case like below.  But it is up to you. 
```
        co_await tmap.for_each_tablet([&] (locator::tablet_id id, const locator::tablet_info& info) -> future<> {
            auto range = tmap.get_token_range(id);
            auto& replicas = info.replicas;
            bool found = false;
            shard_id master_shard_id;

            if (primary_replica_only) {
                auto primary_replica = select_primary_ranges_within_dc ?
                        tmap.get_primary_replica_within_dc(id, erm->get_topology(), mydc) :
                        tmap.get_primary_replica(id);
                if (primary_replica.host == myhostid) {
                    found = true;
                    master_shard_id = primary_replica.shard,
                }
            } else {
                // Repair all tablets belong to this node
                for (auto& r : replicas) {
                    if (r.host == myhostid) {
                        master_shard_id = r.shard;
                        found = true;
                        break;
                    }
                }
            }
            if (found) {
                metas.push_back(repair_tablet_meta{id, range, myhostid, master_shard_id, replicas});
            }
            return make_ready_future<>();
        });
```

Reply:
right, this cross my mind too, but I doubt it's really worth it just to remain closer the the existing implementation.
We can also keep an `optional<tablet_replica>` variable instead of `bool found` and `master_shard_id`. I think that would be more elegant, but again, I'm not sure it's worth the complication in logic.

Reply:
OK. I am good with your version too.